spanner/spanner_snippets/spanner/spanner_opentelemetry_usage.go

```diff
@@ -0,0 +1,130 @@
+//go:build go1.20
+// +build go1.20
+
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+// [START spanner_opentelemetry_usage]
+// Ensure that your Go runtime version is supported by the OpenTelemetry-Go compatibility policy before enabling OpenTelemetry instrumentation.
+// Refer to compatibility here https://github.com/googleapis/google-cloud-go/blob/main/debug.md#opentelemetry
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+
+	"cloud.google.com/go/spanner"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+	"google.golang.org/api/iterator"
+)
+
+func enableOpenTelemetryMetricsAndTraces(w io.Writer, db string) error {
+	// db = `projects/<project>/instances/<instance-id>/database/<database-id>`
+	ctx := context.Background()
+
+	// Create a new resource to uniquely identify the application
+	res, err := newResource()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Enable OpenTelemetry traces by setting environment variable GOOGLE_API_GO_EXPERIMENTAL_TELEMETRY_PLATFORM_TRACING=opentelemetry at the start of the application
+	// Enable OpenTelemetry metrics before injecting meter provider.
+	spanner.EnableOpenTelemetryMetrics()
+
+	// Create a new tracer provider
+	tracerProvider, err := getOtlpTracerProvider(ctx, res)
+	defer tracerProvider.ForceFlush(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Register tracer provider as global
+	otel.SetTracerProvider(tracerProvider)
+
+	// Create a new meter provider
+	meterProvider := getOtlpMeterProvider(ctx, res)
+	defer meterProvider.ForceFlush(ctx)
+
+	// Inject meter provider locally via ClientConfig when creating a spanner client.
```

Comment:
Add the global option in the comment. `// Inject meter provider locally via ClientConfig when creating a spanner client or set via setMeterProvider`

Reply:
Done